sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -250,6 +253,47 @@ export class MessageSender extends LinkEntity {
   private _trySend(encodedMessage: Buffer, sendBatch?: boolean): Promise<void> {
     const sendEventPromise = () =>
       new Promise<void>(async (resolve, reject) => {
+        const actionAfterTimeout = () => {
+          const desc: string =
+            `[${this._context.namespace.connectionId}] Sender "${this.name}" ` +
+            `with address "${this.address}", was not able to send the message right now, due ` +
+            `to operation timeout.`;
+          log.error(desc);
+          const e: AmqpError = {
+            condition: ErrorNameConditionMapper.ServiceUnavailableError,
+            description: desc
+          };
+          return reject(translate(e));
+        };
+
+        const waitTimer = setTimeout(
+          actionAfterTimeout,
+          getRetryAttemptTimeoutInMs(this._getSenderOptions?.retryOptions)
+        );
+        if (!this.isOpen()) {
+          log.sender(
+            "Acquiring lock %s for initializing the session, sender and " +
+              "possibly the connection.",
+            this.senderLock
+          );
+          try {
+            const senderOptions = this._createSenderOptions(
+              getRetryAttemptTimeoutInMs(this._getSenderOptions?.retryOptions)
```

Comment:
Say the user provided timeout to be 60 seconds.
It took `init()` 10 seconds to establish the connection.
Now, the actual send operation should timeout in 50 seconds, but with this code, it will timeout only in 60 seconds.

I recommend we revert the commit https://github.com/Azure/azure-sdk-for-js/pull/7972/commits/46c3b59a815697528ab5c8bbfa3de90d99ea043f

Then, keep track of the time taken for init() to complete.
Subtract that from the timeout value and set that directly on the sender
```
this._sender.sendTimeoutInSeconds = timeoutFromUserOrDefaultValue - timeTakenByInit
```

Reply:
Agree with your points, but with a slight tweak, init should not exceed 60s too.
So, 
1. I'll keep the timeout param for init 
2. And calculate the left over time for the send

Reply:
Then I would say that the outer timer should probably apply only for the init operation, because we have a separate timer inside the `AwaitableSender.send()`

```
if (link not open) {
  start timer
  call init
  close timer
}
this._sender.sendTimeoutInSeconds = timeoutFromUserOrDefaultValue - timeTakenByInit
```

Reply:
Updated with your suggestions. Probably, event-hubs should also be updated?